cmd/lakectl/cmd/common_helpers.go

```diff
@@ -206,6 +210,31 @@ func DieOnResponseError(response interface{}, err error) {
 		DieErr(retrievedErr)
 	}
 }
+func DieOnErrorOrUnexpectedStatusCode(response interface{}, err error, expectedStatusCode int) {
```

Comment:
Could we refactor parts of HTTPResponseAsError with this?  I think it copies some parts.

Reply:
I've tried to, but it seems to be kind of difficult because of the structure of `ResponseAsError` func. All we want in our case is to get the status code from the response when `ResponseAsError` checks and returns an error on its way to get it.  WDYT?